client/consensus/aura/src/lib.rs

```diff
@@ -248,10 +247,15 @@ impl<B, C, E, I, P, Error, SO> sc_consensus_slots::SimpleSlotWorker<B> for AuraW
 	) -> Option<Self::Claim> {
 		let expected_author = slot_author::<P>(slot_number, epoch_data);
 		expected_author.and_then(|p| {
-			self.keystore.read()
-				.key_pair_by_type::<P>(&p, sp_application_crypto::key_types::AURA).ok()
-		}).and_then(|p| {
-			Some(p.public())
+			let has_keys = self.keystore.has_keys(
+				&[(p.to_raw_vec(), sp_application_crypto::key_types::AURA)],
+			);
+			match has_keys {
+				true => {
+					Some(p.clone())
+				},
+				false => None
+			}
```

Comment:
```suggestion
			if self.keystore.has_keys(
				&[(p.to_raw_vec(), sp_application_crypto::key_types::AURA)],
			) {
			    Some(p.clone())
			} else {
			    None
			}
```

I might be missing a way to write this in a more readable way.